rero_ils/modules/selfcheck/api.py

```diff
@@ -0,0 +1,174 @@
+    :returns: ``True`` if patron is authorized or ``False``.
+    """
+    patron = Patron.get_patron_by_barcode(barcode)
+    if patron and patron.is_patron:
+        return authorize_selfckeck_user(patron.get('email'), password)
+    return False
+
+
+def system_status(login, *kwargs):
+    """Selfcheck system status handler.
+
+    Get status of automated circulation system.
+    :param login: selfcheck client login.
+    :returns: The status object.
+    """
+    staffer = Patron.get_patron_by_email(login)
+    return {
+        'institution_id': staffer.get_organisation().get('code')
+    }
+
+
+def enable_patron(barcode, **kwargs):
+    """Selfcheck enable patron handler.
+
+    Enable patron 'barcode' from selfcheck user.
+    :param barcode: patron barcode.
+    :returns: ``True`` if patron exists or ``None``.
+    """
+    # check if invenio_sip2 module is present
+    if check_sip2_module():
+        from invenio_sip2.models import SelfcheckPatronStatus
+
+        patron = Patron.get_patron_by_barcode(barcode)
+        patron_status = SelfcheckPatronStatus()
+        # TODO: add PatronStatusType according account
+        #  e.g.:
+        #   patron_status.add_patron_status_type(
+        #   SelfcheckPatronStatusTypes.CARD_REPORTED_LOST)
+        return {
+            'patron_status': patron_status,
+            'language': patron.get('communication_language', 'und'),
+            'institution_id': patron.get_organisation().get('code'),
+            'patron_id': patron.get('barcode'),
+            'patron_name': patron.formatted_name
+        }
+
+
+def patron_information(barcode, **kwargs):
+    """Get patron information handler.
+
+    Get patron information according 'barcode' from selfcheck user.
+    :param barcode: patron barcode.
+    :returns: The SelfcheckPatronInformation object.
+    """
+    # check if invenio_sip2 module is present
+    if check_sip2_module():
+        from invenio_sip2.models import SelfcheckPatronInformation
+
+        patron = Patron.get_patron_by_barcode(barcode)
+        patron_account_information = SelfcheckPatronInformation(
+            patron_id=barcode,
+            patron_name=patron.formatted_name,
+            patron_email=patron.get('email'),
+            patron_phone=patron.get('phone'),
+            patron_address=format_patron_address(patron),
+            institution_id=patron.get_organisation().get('code'),
+            language=patron.get('communication_language', 'und'),
+            currency_type=patron.get_organisation().get('default_currency')
+        )
+
+        loans = get_loans_by_patron_pid(patron.pid)
+        for loan in loans:
+            item = Item.get_record_by_pid(loan.item_pid)
+            if loan['state'] == 'ITEM_ON_LOAN':
+                patron_account_information.get('charged_items', []).append(
+                    item.get('barcode'))
+                if is_overdue_loan(loan):
+                    patron_account_information.get('overdue_items', [])\
```

Comment:
same here.

Reply:
overdue_items is defined in SelfcheckPatronInformation